ssf: remove the reset_evt sdfw_update and echo service (#19490)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

SDFW samples
------------

> * Removed the ``ssf_client`` sample since all services being shown has been removed.

Reviewer: ```suggestion
* Removed the SDFW: Service Framework Client sample since all related services are removed.
```

Reviewer: Not so sure about the sample removal just like that. Shouldn't it first be stated as deprecated and removed only in later release?

Author: The server side implementation will be removed, so it has to be removed.

Reviewer: Where is that then? Was it declared deprecated before removing it?

Reviewer: @hakonfam You haven't still updated the text according to my suggestion.

SDFW samples
------------

> * Removed the ``ssf_client`` sample since all services being shown has been removed.

Reviewer 2: ```suggestion
* Removed the ``ssf_client`` sample, as all services demonstrated by the sample have been removed.
```

Reviewer: I also have a comment regarding this entry. I thought it was implemented already, but no. ssf_client need to be opened and written as SDFW: Service Framework Client

Author: I joined your suggestions

Other libraries
---------------

> * Removed the unused SDFW services ``echo_service``, ``reset_evt_service`` and ``sdfw_update_service``.

Reviewer: ```suggestion
* Removed the unused SDFW services ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.
```

Reviewer: This entry is a bit suspicious, as we don't seem to mention these services in any of the NCS libraries.

Author: They are sample services that are not used by any other library.

Reviewer: I'd say we can drop this entry.

Other libraries
---------------

> * Removed the unused SDFW services ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.

Reviewer 2: ```suggestion
* Removed the following unused SDFW services: ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.
```

> * Removed the following unused SDFW services: ``echo_service``, ``reset_evt_service``, and ``sdfw_update_service``.

Reviewer 2: Shouldn't this entry be moved in this page? https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/releases_and_maturity/abi_compatibility.html

Author: It will appear there as well when the SDFW version that removes them is released. The comment in this changelog is referring to the removal of the client side implementation of the services.
